contracts/investment/InvestmentAsset.sol

```diff
@@ -133,22 +159,25 @@ contract InvestmentAsset {
         var (currentInvestor, investedValue) = makeAvailable();
         Canceled(_id, owner, currentInvestor, investedValue);
         return true;
-    }    
+    }
 
     // Accept the investor as the asset buyer and withdraw funds
     function withdrawFunds(string _id, bytes _agreementHash)
         onlyOwner
         hasStatus(Status.PENDING_OWNER_AGREEMENT)
+        onlyAgreed(_agreementHash)
         public
         returns(bool)
     {
         // compare the document signed by the offer owner and investor
         if (keccak256(agreementHash) == keccak256(_agreementHash)) {
-            uint256 value = this.balance;            
+            uint256 value = this.balance;
             owner.transfer(value);
             status = Status.INVESTED;
             Withdrawal(_id, owner, investor, value, agreementHash);
             return true;
+        }else {
+            revert();
```

Comment:
If...else statement should not be here. The modifier 'onlyAgreed' alredy does this.